substrate/frame/revive/src/lib.rs

```diff
@@ -512,6 +512,14 @@ pub mod pallet {
 		AddressMapping,
 	}
 
+	#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+	pub enum IncrementOnce {
+		/// Indicates that the nonce has not been incremented yet.
+		No,
+		/// Indicates that the nonce has already been incremented.
+		AlreadyIncremented,
+	}
```

Comment:
This is a better name. Cause we are not actually bumping it depending on the value. We are decrementing it when `Yes`.

```suggestion
	pub enum NonceAlreadyIncremented {
		/// Indicates that the nonce has already been incremented.
		///
		/// This happens when the instantiation is triggered by a transaction.
		Yes,
		/// Indicates that the nonce has not been incremented yet.
		///
		/// This happens when the instantiation is triggered by a dry-run or another contract.
		No,
	}
```

Reply:
sure